### ISSUE TITLE
SARIF mode line and column numbers off by 1

### ISSUE THREAD

**Issue by rnikoopour** [mgechev/revive#771]
**Describe the bug**
When outputting SARIF mode column and line numbers are both off by 1.

When uploading the SARIF file to Github, this results in Github misplacing the SARIF information in the UI.

**To Reproduce**
1. Create `foo.go` with the following contents:
    ```go
    package main
    
    const FOO_BAR = "Testing"

    func main() {
            accountId = "Needs fixing"
    }
    ```
2. Run `revive -formatter sarif foo.go`
3. Look at the sarif output and see column and line numbers are off by one (Example finding below)
    ```json
    {
      "level": "warning",
      "locations": [
        {
          "physicalLocation": {
            "artifactLocation": {
              "uri": "foo.go"
            },
            "region": {
              "startColumn": 6,
              "startLine": 2
            }
          }
        }
      ],
      "message": {
        "text": "exported const FOO_BAR should have comment or be unexported"
      },
      "ruleId": "exported"
    }
    ```

**Expected behavior**
Formatter for sarif should not line numbers and columns off by one

**Logs**
Full sarif output for `foo.go`
```json
{
  "runs": [
    {
      "results": [
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "foo.go"
                },
                "region": {}
              }
            }
          ],
          "message": {
            "text": "should have a package comment"
          },
          "ruleId": "package-comments"
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "foo.go"
                },
                "region": {
                  "startColumn": 6,
                  "startLine": 2
                }
              }
            }
          ],
          "message": {
            "text": "exported const FOO_BAR should have comment or be unexported"
          },
          "ruleId": "exported"
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "foo.go"
                },
                "region": {
                  "startColumn": 6,
                  "startLine": 2
                }
              }
            }
          ],
          "message": {
            "text": "don't use ALL_CAPS in Go names; use CamelCase"
          },
          "ruleId": "var-naming"
        }
      ],
      "tool": {
        "driver": {
          "informationUri": "https://revive.run",
          "name": "revive",
          "rules": [
            {
              "helpUri": "https://revive.run/r#package-comments",
              "id": "package-comments",
              "properties": {
                "severity": "warning"
              }
            }
          ]
        }
      }
    }
  ],
  "version": "2.1.0"
}
```

**Desktop (please complete the following information):**
 - OS: alpine container
 - Version of Go: 1.19

**Additional context**
I believe this is related to this subtractions these two lines:

https://github.com/mgechev/revive/blob/04728cf0de86ab99fb3327a1ea48f7d51a411352/formatter/sarif.go#L84-L85
